core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -383,13 +383,17 @@ public void sendMessage(ReqT message) {
     Preconditions.checkState(stream != null, "Not started");
     Preconditions.checkState(!cancelCalled, "call was cancelled");
     Preconditions.checkState(!halfCloseCalled, "call was half-closed");
+    InputStream messageIs = null;
     try {
-      // TODO(notcarl): Find out if messageIs needs to be closed.
-      InputStream messageIs = method.streamRequest(message);
+      messageIs = method.streamRequest(message);
```

Comment:
A minor note: if the consumer retained a ref to this IS we might accidentally close it prematurely.  For example, they may push it to some queue for asynchronous processing.   

I don't *think* thats a problem, but it is an API change.   This probably warrants a javadoc comment in the MethodDecriptor.

Reply:
Done